app/pathogen/arbovirus/dashboard/(visualizations)/country-seroprevalence-comparison-scatter-plot.tsx

```diff
@@ -0,0 +1,204 @@
+};
+
+export const CountrySeroprevalenceComparisonScatterPlot = () => {
+  const state = useContext(ArboContext);
+
+  const stateFilteredDataWithEstimateNumbers = useMemo(
+    () =>
+      state.filteredData
+        .filter(
+          (dataPoint) =>
+            dataPoint.seroprevalenceCalculated95CILower &&
+            dataPoint.seroprevalenceCalculated95CIUpper
+        )
+        .sort(
+          (dataPointA, dataPointB) =>
+            dataPointA.seroprevalence - dataPointB.seroprevalence
+        )
+        .map((dataPoint, index) => ({
+          ...dataPoint,
+          estimateNumber: index + 1,
+        }))
+        .map((dataPoint) => ({
+          ...dataPoint,
+          seroprevalence: parseFloat(
+            (dataPoint.seroprevalence * 100).toFixed(1)
+          ),
+          seroprevalenceCalculated95CILower: parseFloat(
+            (dataPoint.seroprevalenceCalculated95CILower * 100).toFixed(1)
+          ),
+          seroprevalenceCalculated95CIUpper: parseFloat(
+            (dataPoint.seroprevalenceCalculated95CIUpper * 100).toFixed(1)
+          ),
+          seroprevalenceError: [
+            parseFloat(
+              (
+                dataPoint.seroprevalence * 100 -
+                dataPoint.seroprevalenceCalculated95CILower * 100
+              ).toFixed(1)
+            ),
+            parseFloat(
+              (
+                dataPoint.seroprevalenceCalculated95CIUpper * 100 -
+                dataPoint.seroprevalence * 100
+              ).toFixed(1)
+            ),
+          ],
+        })),
+    [state.filteredData]
+  );
+
+  const { rechartsData } = useMemo(
+    () =>
+      groupDataForRecharts({
+        data: stateFilteredDataWithEstimateNumbers,
+        primaryGroupingFunction: (dataPoint) =>
+          dataPoint.pathogen as arbovirusesSF,
+        primaryGroupingSortFunction: (pathogenA, pathogenB) =>
+          pathogenA !== pathogenB ? (pathogenA < pathogenB ? -1 : 1) : 0,
+        secondaryGroupingFunction: (dataPoint) => dataPoint.country as string,
+        secondaryGroupingSortFunction: (countryA, countryB) =>
+          countryA !== countryA ? (countryA < countryB ? -1 : 1) : 0,
+        transformOutputValue: (data) => data,
+      }),
+    [stateFilteredDataWithEstimateNumbers]
+  );
+
+  const pathogensAvailable = useMemo(
+    () => uniq(rechartsData.map((dataPoint) => dataPoint.primaryKey)),
+    [rechartsData]
+  );
+  const countriesAvailable = useMemo(
+    () =>
+      uniq(
+        rechartsData.flatMap((dataPoint) =>
+          typedObjectKeys(dataPoint).filter((value) => value !== "primaryKey")
+        )
+      ),
+    [rechartsData]
+  );
+
+  if (pathogensAvailable.length > 1) {
+    return (
+      <div>
+        <p>
+          This graph cannot be shown when the data contains more than one
+          arbovirus. If you&apos;re interested in generating this graph, please use
+          the filter to the left to filter out data.
+        </p>
+      </div>
+    );
+  }
```

Comment:
Image below. Looks a little bit silly but this is the best way I could come up with of instructing the user that this graph exists and what they need to do to make it appear. I think when the graphs are collapsible we'll make the heights more dynamic and this will look a bit less silly. It's at the bottom right now because of this but I think it would be cool at some point in the future to re-arrange this graph closer to the top if it is populated but that's for another time :)

Copy is possibly a bit questionable too so that's something we can get feedback on from Thursday,

![image](https://github.com/serotracker/dashboard/assets/86806388/4b48e00c-44d6-41a4-8118-f9d908f56aed)